flow-typed/box-annotations.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @flow
+ * @file Flow types
+ * @author Box
+ */
+/* eslint-disable no-use-before-define */
+import type { MessageDescriptor, InjectIntlProvidedProps } from 'react-intl';
+
+type AnnotationPermissions = {
+    can_edit: boolean,
+    can_delete: boolean
+};
+
+type Coordinates = {
+    x: Number,
+    y: Number
+};
+
+// [x1, x2, x3, x4, y1, y2, y3 ,y4]
+type QuadPoint = {
+    x1: Number,
+    y1: Number,
+    x2: Number,
+    y2: Number,
+    x3: Number,
+    y3: Number,
+    x4: Number,
+    y4: Number
+};
+type QuadPoints = Array<QuadPoint>;
+
+type Path = Array<Coordinates>;
+
+type Location = {
+    min: Coordinates,
+    max: Coordinates,
+    dimensions: {
+        x: Number,
+        y: Number
+    },
+    page: Number
+};
+
+type PointLocation = {
+    coordinates: Coordinates
+} & Location;
+
+type HighlightLocation = {
+    coordinates: QuadPoints
+} & Location;
```

Comment:
Would highlights need `dimensions` from `Location` if they already have their boundary points in `coordinates`? It seems like height/width should be derived rather than stored.